pkg/api/v1/defaults_test.go

```diff
@@ -232,55 +232,418 @@ func TestSetDefaultReplicationControllerReplicas(t *testing.T) {
+			got.LivenessProbe.SuccessThreshold != expected.LivenessProbe.SuccessThreshold ||
+			got.LivenessProbe.PeriodSeconds != expected.LivenessProbe.PeriodSeconds ||
+			got.LivenessProbe.TimeoutSeconds != expected.LivenessProbe.TimeoutSeconds {
+			return fmt.Errorf("different LivenessProbe: got <%v>, expected <%v>", got.LivenessProbe, expected.LivenessProbe)
+		}
+
+		// Assert ReadinessProbe
+		if got.ReadinessProbe.Handler.HTTPGet.Path != expected.ReadinessProbe.Handler.HTTPGet.Path ||
+			got.ReadinessProbe.Handler.HTTPGet.Scheme != expected.ReadinessProbe.Handler.HTTPGet.Scheme ||
+			got.ReadinessProbe.FailureThreshold != expected.ReadinessProbe.FailureThreshold ||
+			got.ReadinessProbe.SuccessThreshold != expected.ReadinessProbe.SuccessThreshold ||
+			got.ReadinessProbe.PeriodSeconds != expected.ReadinessProbe.PeriodSeconds ||
+			got.ReadinessProbe.TimeoutSeconds != expected.ReadinessProbe.TimeoutSeconds {
+			return fmt.Errorf("different ReadinessProbe: got <%v>, expected <%v>", got.ReadinessProbe, expected.ReadinessProbe)
+		}
+
+		return nil
+	}
+
+	assertLifeCycle := func(got, expected *v1.Container) error {
+		if got.Lifecycle.PostStart.HTTPGet.Path != expected.Lifecycle.PostStart.HTTPGet.Path ||
+			got.Lifecycle.PostStart.HTTPGet.Scheme != expected.Lifecycle.PostStart.HTTPGet.Scheme {
+			return fmt.Errorf("different LifeCycle: got <%v>, expected <%v>", got.Lifecycle, expected.Lifecycle)
+		}
+
+		return nil
+	}
+
+	cpu, _ := resource.ParseQuantity("100Gi")
+	mem, _ := resource.ParseQuantity("100Mi")
 
 	tests := []struct {
-		rc               v1.ReplicationController
-		expectPullPolicy v1.PullPolicy
+		name       string
+		rc         v1.ReplicationController
+		expected   []v1.Container
+		validators []InitContainerValidator
 	}{
 		{
+			name: "imagePullIPolicy",
+			rc: v1.ReplicationController{
+				Spec: v1.ReplicationControllerSpec{
+					Template: &v1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Annotations: map[string]string{
+								"pod.beta.kubernetes.io/init-containers": `
+                                [
+                                    {
+                                        "name": "install",
+                                        "image": "busybox"
+                                    }
+                                ]`,
+							},
+						},
+					},
+				},
+			},
+			expected: []v1.Container{
+				{
+					ImagePullPolicy: v1.PullAlways,
+				},
+			},
+			validators: []InitContainerValidator{assertImagePullPolicy},
+		},
+		{
+			name: "FieldRef",
+			rc: v1.ReplicationController{
+				Spec: v1.ReplicationControllerSpec{
+					Template: &v1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Annotations: map[string]string{
+								"pod.beta.kubernetes.io/init-containers": `
+                                [
+                                    {
+                                    "Name": "fun",
```

Comment:
these are incorrect... there should never be upper-case keys in API json objects... did you serialize internal API objects to obtain these?